File: selection_sort.py
#Write a selection sort to sort the algo this data [43, 24, 45, 27, 35, 47, 22, 48]
#Approach the problem using an encapsulation techiques

def exchangeItem(arr, indexA, indexB):
    temp = arr[indexA]
    arr[indexA] = arr[indexB]
    arr[indexB] = temp

def findMin(arr, startIndex):
    minVal = arr[startIndex]
    minIndx = startIndex
    for i in range(startIndex + 1, len(arr)):
        print(i)
        if minVal > arr[i]:
            minVal = arr[i]
            minIndx = i
    return minIndx

def selection_sort(arr):
 
    for i in range(len(arr)):
        minIndx = findMin(arr, i)
        exchangeItem(arr, i, minIndx )


arr = [43, 24, 45, 27, 35, 47, 22, 48]
print(findMin(arr, 4))
print(arr)

File: js_algo/selectionSort.js
function swapItem(arr, indexA, indexB){
    let temp = arr[indexA]
    arr[indexA] = arr[indexB]
    arr[indexB] = temp
}

function findMin(arr, startIndex){
    let minval = arr[startIndex]
    let minIndx = startIndex
    for(let i = startIndex + 1; i < arr.length; i++ ){
        if(minval > arr[i]){
            minval = arr[i]; // To update the whole arr to arrangfe in decending order just alter this line to minVal  < arr[i]
            minIndx = i
        }
    }

    return minIndx;
}

function selectionSort(arr){
    for (let i = 0; i < arr.length; i++){
        let minIndx = findMin(arr, i)
        console.log(minIndx)
        swapItem(arr, i, minIndx)
    }
}


const arr = [43, 24, 45, 27, 35, 47, 22, 48]

selectionSort(arr)
//swapItem(arr, 0, 7)
//console.log(findMin(arr, 6))
console.log(arr)
